### ISSUE TITLE
Geolocation API always turns on

### ISSUE THREAD

**Issue by Fritz66** [BrowserWorks/waterfox#3441]
### What happened?

I prefer to disable the Geolocation API, but everytime I start Waterfax it's enabled again.
Is it only me experiencing this issue?
Version G6.0.13

### Reproducible?

- [ ] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

G6

### What platform are you seeing the problem on?

Linux

**Comment by MrAlex94** [comment]
How are you disabling it? I was under the impression geo didn't work on Linux.

**Comment by Fritz66** [comment]
I tried it via **about:preferences#general**, but I noticed that there is no switch available under **about:config**.
In order to disable it I set the URL to about:blank.

https://freeimghost.net/images/2024/05/15/Bildschirmfoto-zu-2024-05-15-17-07-43.png

**Comment by kubatron** [comment]
> I tried it via **about:preferences#general**, but I noticed that there is no switch available under **about:config**. In order to disable it I set the URL to about:blank.
> 
> https://freeimghost.net/images/2024/05/15/Bildschirmfoto-zu-2024-05-15-17-07-43.png

Just go to **about:config** -> **geo.enabled** -> and change to **false**

it is very weird that Waterfox has some issue with GUI option that are not able to change like API Geo, OSCP Queries and that should be fix in my opinion.

**Comment by Fritz66** [comment]
@kubatron
Did you read my OP?
Please re-read the first sentence.

**Comment by LAURAilway** [comment]
I poked around the preferences config trying to fix this.
Turns out, this is more or less intended behavior.
The radio buttons for the geo api don't actually change the `geo.enabled` setting, as one might assume, they change `geo.provider.network.url`, toggling between "https://location.services.mozilla.com/v1/geolocate?key=%MOZILLA_API_KEY%" and "".
This is probably to facilitate the default value for `geo.provider.network.url` being "", as to prevent any calls to Mozilla that aren't strictly necessary.
The buttons, however, seem to be struggling with detecting the current setting, therefore making it appear as if the setting itself did reset on restart.

tl;dr: the setting works as intended, this is simply a ui problem

**Comment by Fritz66** [comment]
@LAURAilway , thanks for your reply.

Just curious: does **about:blank** the same job as **""** ?

**Comment by LAURAilway** [comment]
It should, however, the setting that actually controls the geo api url isn't `browser.geolocation.warning.infoURL`, but `geo.provider.network.url`, which is actually already being changed by the settings page.
(The setting you changed in about:config seems to just configure the information page for the geo api, you can try that yourself by visiting the original url [https://www.mozilla.org/%LOCALE%/firefox/geolocation/] and replacing %LOCALE% with any locale). 
Setting `geo.provider.network.url` to about:blank should do the same as leaving it empty tho.

**Comment by Fritz66** [comment]
The funny thing is that **geo.provider.network.url** was already empty, but the Geolocation API settings still show it's enabled: https://postimg.cc/5jbYmqdc
Am I right that I can just ignore the Geolocation API settings as long as **geo.provider.network.url** is empty?

"Turns out, this is more or less intended behavior."
Well, maybe the developers will rethink that.

**Comment by LAURAilway** [comment]
Yes, you can ignore it if `geo.provider.network.url` is empty.
As stated above, this is just a bug with the display, the functionality works as intended (that is what I meant with intended behavior).
